### PR TITLE
feat(project): NG_/LRG in-place c./n.→g. projection for compound alleles and UTR coordinates

### LINKED ISSUE #851: feat(project): complete NG_/LRG_ in-place c./n.→g. projection for compound alleles and UTR coordinates
Part of #326 (genomic axis; spec-arbitration sweep 2026-06-25).

For an `NG_(NM_)` genomic-context input, single-variant `c.→NG_:g.` projection already works, but two sub-cases fail:

1. **Compound alleles are not projected.** `c.[…]` silently returns the un-projected `c.` form instead of `g.[…]`.
   - `NG_012337.1(NM_003002.2):c.[274G>T;278A>G]` → expected `NG_012337.1:g.[7125G>T;7129A>G]`, got `…(NM_003002.2):c.[274G>T;278A>G]`.
2. **5'UTR coordinates are hard-rejected** even though the `NG_` parent supplies the 5' flank.
   - `NG_012337.1(NM_012459.2):c.-35_*1del` → expected `NG_012337.1:g.3448_4957del`, got `err: Invalid CDS position: -35`.

Spec: `numbering.md` L41 (the `NG_(NM_):c.→g.` construction is well-defined), `refseq.md` L46 (the `NG_` reference supplies UTR/flank context). Extend genomic projection over allele members and resolve negative-UTR coordinates under an `NG_` parent. ~7 genomic-axis rows.

### PR DESCRIPTION
**Pull request by nh13**
Closes #851. Part of #326.

For an `NG_(NM_)`-genomic-context input, completes in-place `c.→NG_/LRG_:g.` projection for two previously-failing sub-cases, clearing all 6 `NG_012337.1` genomic-axis rows.

**Fixes (3 rows):**
- **Route compound alleles** through `project_to_genomic` — each member is projected into the parent genomic frame and reassembled (all-or-nothing), e.g. `c.[274G>T;278A>G]` → `g.[7125G>T;7129A>G]`. Previously alleles fell through unprojected.
- **Genomic-order sort** of projected members (`alleles.md` §"listed in genomic order") — a minus-strand transcript projects members in descending order; cis alleles are re-sorted ascending (trans alleles preserve haplotype order).
- **5'UTR parent-flank extension** — a `c.-N` position upstream of the modeled transcript start now extends linearly into the contiguous NG_/LRG_ flank (`refseq.md`) instead of erroring, e.g. `c.-35_*1del` → `g.3448_4957del`.

**Annotations (3 rows):** the `=`-member rows (`c.[274;600]`, `c.[274=;275del]`, `c.[274del;275=]`) are dispositioned `accepted_divergence` on the genomic axis under the existing `ferro-policy-654-explicit-identity-rendering`: ferro retains the explicit `=` no-change form (`substitution.md`; bare/dropped allele members are non-spec per `syntax.yaml`), mutalyzer drops/bares them.

Out of scope (verified, deferred elsewhere): legacy/gene-symbol selectors (#858), `NG_009299.1` re-anchor (#872), minus-strand single-variant repeats.

CI-runnable MockProvider tests in `tests/it/issue_851_compound_utr_projection.rs` cover routing, sort (with trans no-reorder guard), 5'UTR flank (minus + plus + within-UTR boundary), the `=`-member form, and all-or-nothing decline.

## Summary by CodeRabbit

* **New Features**
  * Enhanced genomic projection for compound alleles: cis members are re-sorted by genomic start while trans member order is preserved.
  * Improved 5’UTR mapping for non-intronic CDS inputs, including flanks that extend beyond transcript boundaries (with clear rejection when outside the placed span).
* **Bug Fixes**
  * Improved handling for allele inputs already in genomic form (idempotent projection/normalization).
  * Refined normalization behavior for compound alleles with bare positions and explicit no-change (`=`) members.
* **Tests**
  * Added Issue 851 integration coverage (plus/minus strands, ordering semantics, and 5’UTR edge cases).
  * Updated mutalyzer normalization fixtures and divergence-acceptance baselines.

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

File: src/data/mapping.rs [hunk 1]
```diff
@@ -359,6 +359,53 @@ impl CoordinateMapper {
...
+                        .tx_to_genome(0)
+                        .ok_or_else(|| FerroError::InvalidCoordinates {
+                            msg: format!(
+                                "cannot anchor 5' flank for {transcript_id} at transcript base 0"
+                            ),
+                        })?;
+                    let genome_pos = match tx.strand {
+                        Strand::Minus => g0 + under,
+                        Strand::Plus => {
+                            g0.checked_sub(under)
```
[the remark this answers is not in the document]
Comment (nh13) [reply src/data/mapping.rs]: Done — `offset` now uses `checked_neg` (guards `i64::MIN`) and the minus-strand extension uses `checked_add`, both returning `InvalidCoordinates` on overflow, matching the plus-strand `checked_sub` sibling.

File: src/data/mapping.rs [hunk 2]
```diff
@@ -359,6 +359,53 @@ impl CoordinateMapper {
...
+                let offset = (-cds_pos.base) as u64;
+                if offset > cds_start {
+                    let under = offset - cds_start;
+                    let g0 = tx
+                        .tx_to_genome(0)
+                        .ok_or_else(|| FerroError::InvalidCoordinates {
+                            msg: format!(
+                                "cannot anchor 5' flank for {transcript_id} at transcript base 0"
+                            ),
+                        })?;
```
[the remark this answers is not in the document]
Comment (nh13) [reply src/data/mapping.rs]: Done — the flank anchor at transcript base 0 now runs the same CIGAR-insertion-gap guard used on the non-intronic path below (`exon_for_tx_pos` + `cigar_insertion_gap_at_tx_pos`), returning `AlignmentGap` before `tx_to_genome(0)` so an anchor inside a transcript-only insertion no longer fabricates a coordinate.

File: tests/it/issue_851_compound_utr_projection.rs [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        ManeStatus::default(),
+        None,
+        None,
+    ));
+    let seq = format!(
+        "{}{}{}",
+        "N".repeat(1999),
+        "ACGTACGTACGTACGTACGT",
+        "N".repeat(80)
+    );
```
[the remark this answers is not in the document]
Comment (nh13) [reply tests/it/issue_851_compound_utr_projection.rs:183]: Done — `minus_fixture()` now seeds deterministic flank bases (g.2020 = `T`), and both minus-strand regressions assert the full string exactly: `NG_M.1:g.2019T>G` and `NG_M.1:g.2020T>G`. These pin accession, edit, and the minus→plus reverse-complement (A>C → T>G).

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
The out-of-bounds case is already rejected in the caller that holds placement context: `reanchor_genome_output` → `reanchor_genome_to_parent` declines via `GenomicPlacement::nc_to_parent`, which returns `None` for any NC coordinate outside `[nc_start, nc_end]` (the placed span), yielding `UnsupportedProjection: ... an endpoint falls outside the placed genomic span`. A large 5'-flank `c.-N` whose NC coordinate leaves the placement therefore declines rather than emitting a fabricated coordinate. Added plus- and minus-strand `issue_851` regressions (`plus_strand_5utr_flank_outside_placement_declines`, `minus_strand_5utr_flank_outside_placement_declines`) that drive a flank past `nc_start`/`nc_end` and assert the decline.

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review
